src/transformers/models/omdet_turbo/modeling_omdet_turbo.py

```diff
@@ -0,0 +1,1916 @@
+        encoder_features = self.encoder(image_features)[0]
+        label_features, prompt_features, prompt_mask = self.get_language_embedding(labels, tasks)
+        decoder_outputs = self.decoder(encoder_features, label_features, prompt_features, prompt_mask)
+
+        return OmDetTurboModelOutput(
+            decoder_bboxes=decoder_outputs.decoder_bboxes,
+            decoder_cls=decoder_outputs.decoder_cls,
+            encoder_bboxes=decoder_outputs.encoder_bboxes,
+            encoder_cls=decoder_outputs.encoder_cls,
+            encoder_last_hidden_state=encoder_features,
+            intermediate_reference_points=decoder_outputs.intermediate_reference_points,
+        )
+
+
+@dataclass
+class OmDetTurboObjectDetectionOutput(ModelOutput):
+    """
+    Base class for outputs of the OmDetTurbo model.
+
+    Args:
+        decoder_bboxes (`torch.FloatTensor` of shape `(batch_size, num_queries, num_classes, 4)`):
+            The predicted bounding boxes of the objects.
+        decoder_cls (`torch.FloatTensor` of shape `(batch_size, num_queries, num_classes)`):
+            The predicted classes of the objects.
+        encoder_bboxes (`torch.FloatTensor` of shape `(batch_size, num_queries, num_classes, 4)`):
+            The predicted bounding boxes of the objects from the encoder.
+        encoder_cls (`torch.FloatTensor` of shape `(batch_size, num_queries, num_classes)`):
+            The predicted classes of the objects from the encoder.
+        encoder_last_hidden_state (`torch.FloatTensor` of shape `(batch_size, num_queries, hidden_size)`):
+            The last hidden state of the encoder.
+        intermediate_reference_points (`torch.FloatTensor` of shape `(batch_size, num_queries, 2)`:
+            The intermediate reference
+    """
+
+    decoder_bboxes: torch.FloatTensor = None
+    decoder_cls: torch.FloatTensor = None
+    encoder_bboxes: torch.FloatTensor = None
+    encoder_cls: Optional[Tuple[torch.FloatTensor]] = None
+    encoder_last_hidden_state: Optional[torch.FloatTensor] = None
+    intermediate_reference_points: Optional[Tuple[Tuple[torch.FloatTensor]]] = None
+
+
+@add_start_docstrings(
+    """
+    OmDetTurbo Model (consisting of a vision and a text backbone, and encoder-decoder architecture) outputting
+    bounding boxes and classes scores for tasks such as COCO detection.
+    """,
+    OMDET_TURBO_START_DOCSTRING,
+)
+class OmDetTurboForObjectDetection(OmDetTurboPreTrainedModel):
+    def __init__(self, config: OmDetTurboConfig):
+        super().__init__(config)
+        # Create backbone
+        self.backbone = OmDetTurboVisionBackbone(config)
+        self.language_backbone = OmDetTurboLanguageBackbone(config=config)
+        self.encoder = OmDetTurboHybridEncoder(config)
+        self.decoder = OmDetTurboDecoder(config)
+        self.num_queries = config.num_queries
+
+        self.language_cache_label = OmDetTurboLRUCache(100)
+        self.language_cache_prompt = OmDetTurboLRUCache(100)
```

Comment:
Thinking about this more, we might have to make it a static method and pass in the backbone as an arg. I'm not sure how the cache checks the objects, but it's possible `self` will be changing overtime, removing the value of the caching behaviour

Reply:
However it seems that this method would not work with batch inference. Right now we check each embeddings to see if they're in the cache, and add the ones that are not to a batch then use batch inference. Since python lru cache doesn't return if there was a cache hit or not (from what I've read at least) we wouldn't be able to do that? 
The only implementation I see with python lru cache if we want to use batch inference is either no embeddings have changed and we get a cache it, or we recompute all embeddings if any of them has changed.